yaml/semgrep/consistency/generate_rules.py

```diff
@@ -0,0 +1,56 @@
+from pdb import runeval
+import requests
+import yaml
+import os
+
+def fetch_lang_data():
+    print('fetching latest lang configs..')
+    lang_url = "https://raw.githubusercontent.com/returntocorp/semgrep-langs/main/lang.json"
+    return requests.get(lang_url).json()
+
+def generate_rule(lang_data):
+    alt_names = list(filter(lambda x: x != lang_data['id'], lang_data['keys']))
+    rule_obj = None
+    if len(alt_names) > 0:
+        key_string = '|'.join(alt_names)
+        rule_obj = {
+            "rules":[
+                {
+                    "id":f"lang-consistency-{lang_data['id']}",
+                    "patterns":[
+                        {
+                            "pattern" : "$X"
+                        },
+                        {"pattern-inside":"languages: ..."},
+                        {
+                            "metavariable-regex": {
+                                "metavariable":"$X",
+                                "regex":f"^({key_string})$"
+                            }
+                        }
+                    ],
+                    "message": "Found '$X' in language config which diverges from semgrep.dev normalization.",
```

Comment:
Maybe say: `"$X is a valid langauge key but for standardization please use " + lang_data['id']"` just a bit clearer to me even though we already have `fix`

Reply:
fair point will fix